### DISCUSSION TITLE
Sum over ct

### DISCUSSION THREAD

**Discussion by jansteinhauser** [magpiemodel/magpie#299] [Q&A] [answered]
Hello again,

another small sanity check I'd like your input on. Here and there, MAgPIE sums over `ct`, e. g. in the [bioenergy module](https://rse.pik-potsdam.de/doc/magpie/4.3/60_bioenergy.htm):
`q60_bioenergy(i2,kall) ..
 vm_dem_bioen(i2,kall) * fm_attributes("ge",kall) =g=`
` sum(ct, i60_1stgen_bioenergy_dem(ct,i2,kall)) +
      v60_2ndgen_bioenergy_dem_dedicated(i2,kall) +
      v60_2ndgen_bioenergy_dem_residues(i2,kall);`

As I understand it, current time `ct` always contains just one set entry, the current time step, as looped through in the optimization module. As such, this sum is only there to get rid of the time dimension. So, effectively,  `vm_dem_bioen(i2,kall)` in this example doesn't have a time dimension but is calculated for every time step as part of the over-arching optimization loop.

Is that correct or am I missing something?

Thank you and all the best!
Jan

**Comment by abhimishr** [comment, marked as answer]
That is correct. All interface variables vm or any other vXX variable contains the optimization step value which is passed on to ov or ovXX variables in postsolve.

**Reply by jansteinhauser** [reply]
Thanks for clarifying. Enjoy your weekend!
